lib/dolly/document.rb

```diff
@@ -143,9 +143,16 @@ def self.property *ary
       options           = ary.pop if ary.last.kind_of? Hash
       options         ||= {}
 
-      ary.each do |name|
+      if ary.count==1 && options[:class_name] == Hash && block_given?
+        name = ary.first
         self.properties[name] = Property.new options.merge(name: name)
         self.write_methods name
+        yield self.properties[name]
+      else
+        ary.each do |name|
+          self.properties[name] = Property.new options.merge(name: name)
+          self.write_methods name
```

Comment:
Will do, hadn't looked into refactoring for DRYness yet, currently I was just exploring whether or not it is the correct approach